Sources/MessagingPush/PushClickHandler.swift

```diff
@@ -0,0 +1,189 @@
+import CioInternalCommon
+import CioTracking
+import Foundation
+import UserNotifications
+
+protocol PushClickHandler: AutoMockable {
+    func pushClicked(_ response: UNNotificationResponse) -> CustomerIOParsedPushPayload?
+    func setupClickHandling()
+}
+
+// Make class a singleton to avoid being garbage collected. A strong reference needs to be held of instance.
+// We are setting this class to be UNUserNotificationCenter.delegate instance and delegates are usually weak.
+//
+// sourcery: InjectRegister = "PushClickHandler"
+// sourcery: InjectSingleton
+class PushClickHandlerImpl: NSObject, PushClickHandler, UNUserNotificationCenterDelegate {
```

Comment:
New class added which encapsulates all push click behavior of the SDK. This includes the automated push click handling setup as well as what to do once a push does get clicked.